python/metashape_workflow_functions.py

```diff
@@ -155,16 +296,363 @@ def read_yaml(self):
+                # Map to project section
+                if 'project' not in modified_override:
+                    modified_override['project'] = {}
+                modified_override['project'][key] = value
+            else:
+                # Keep other overrides at top level
+                modified_override[key] = value
+
         # Update any of the fields in the override dict to that value
-        self.cfg = recursive_update(self.cfg, override_dict)
+        self.cfg = recursive_update(self.cfg, modified_override)
 
     #### Functions for each major step in Metashape
 
+    def run_step(self, step_name):
+        """
+        Run a single processing step.
+
+        For the 'setup' step, creates a new project (overwriting if exists).
+        For all other steps, loads the existing project first.
+
+        Args:
+            step_name (str): Name of the step to run. Valid steps: setup, match_photos,
+                align_cameras, build_depth_maps, build_point_cloud, build_mesh,
+                build_dem_orthomosaic, match_photos_secondary, align_cameras_secondary, finalize
+
+        Raises:
+            ValueError: If step_name is invalid or prerequisites not met
+        """
+        valid_steps = [
+            "setup",
+            "match_photos",
+            "align_cameras",
+            "build_depth_maps",
+            "build_point_cloud",
+            "build_mesh",
+            "build_dem_orthomosaic",
+            "match_photos_secondary",
+            "align_cameras_secondary",
+            "finalize",
+        ]
+
+        if step_name not in valid_steps:
+            raise ValueError(
+                f"Invalid step name: '{step_name}'. Valid steps are: {', '.join(valid_steps)}"
+            )
+
+        if step_name == "setup":
+            # Setup step creates new project (may overwrite existing)
+            self.validate_prerequisites(step_name)
+            self.setup()
+        else:
+            # All other steps load existing project first
+            self.load_existing_project()
+            self.validate_prerequisites(step_name)
+            method = getattr(self, step_name)
+            method()
+
+    def load_existing_project(self):
+        """
+        Load existing project for step-based execution.
+
+        Constructs the expected project file path and loads it. Also sets up
+        instance variables needed for logging and benchmarking.
+
+        Raises:
+            ValueError: If project file doesn't exist
+        """
+        # Construct the expected project file path (same logic as project_setup)
+        project_name = self.cfg["project"]["project_name"]
+        if project_name == "from_config_filename" or project_name == "":
+            file_basename = os.path.basename(self.config_file)
+            project_name, _ = os.path.splitext(file_basename)
+
+        project_file = os.path.join(self.cfg["project"]["project_path"], ".".join([project_name, "psx"]))
```

Comment:
nit: I know I'm arguing with AI on meaningless style but I'd use an fstring here for formatting the project name for better readability.